sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java

```diff
@@ -229,7 +230,9 @@ public void populateDisplayData(DisplayData.Builder builder) {
 
     private class Reader extends BoundedReader<ValueWithRecordId<T>> {
       private long recordsRead = 0L;
-      private Instant endTime = Instant.now().plus(getMaxReadTime());
+
+      @Nullable private Instant endTime = Instant.now().plus(getMaxReadTime());
```

Comment:
Does plus(null) work?

Reply:
It turns out that it does "work": ["If the amount is zero or null, then this is returned."](http://joda-time.sourceforge.net/apidocs/org/joda/time/Instant.html#plus(org.joda.time.ReadableDuration)). Certainly not the spec I would have chosen. Of course, if it didn't work we would have crashes already.

On the other hand, this initial value is always overwritten in the constructor in a way that contradicts the logic here. So I've just removed this.